src/components/Header/Header.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+type Props = {
+  todos: Todo[],
+  addNewTodo: (todoTitle: string) => Promise<void>,
+  setErrorMessage: (errorMessage: ErrorMessage) => void,
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  addNewTodo,
+  setErrorMessage,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const allTodosCompleted = todos.every(todo => todo.completed);
+
+  const handleFormSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim().length) {
+      setErrorMessage(ErrorMessage.TITLE);
+      setTodoTitle('');
+
+      return;
+    }
+
+    try {
+      setIsInputDisabled(true);
+      await addNewTodo(todoTitle);
+    } catch {
+      setErrorMessage(ErrorMessage.ADD);
+    } finally {
+      setIsInputDisabled(false);
+      setTodoTitle('');
```

Comment:
reset the input right away, because it keeps the text until the request is ended now, looks a little strange